### PR TITLE
Add segmentation mask support to Mosaic and JitteredResize

### PR DESCRIPTION
**Pull request by ianstenbit**
Also fixed the `keras` import on a few other layers.

### CODE REVIEW

File: examples/layers/preprocessing/segmentation/mosaic_demo.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+"""
+import demo_utils
+import tensorflow as tf
+
+from keras_cv.layers import preprocessing
+
+
+def main():
+    ds = demo_utils.load_oxford_iiit_pet_dataset()
+    jittered_resize = preprocessing.Mosaic()
```
Comment (jbischof) [inline examples/layers/preprocessing/segmentation/mosaic_demo.py:28]: `s/jittered_resize/mosaic`?

File: keras_cv/layers/preprocessing/jittered_resize.py [hunk 2]
```diff
@@ -205,6 +205,26 @@ def augment_images(self, images, transformations, **kwargs):
         )
         return tf.cast(scaled_images, self.compute_dtype)
 
+    def augment_segmentation_masks(
```
Comment (jbischof) [inline keras_cv/layers/preprocessing/jittered_resize.py:208]: Why can't we call `self.augment_images` here?

**Review by jbischof — APPROVED** [review]
Thanks!

File: examples/layers/preprocessing/segmentation/mosaic_demo.py (hunk 1 above)
Comment (ianstenbit) [reply examples/layers/preprocessing/segmentation/mosaic_demo.py]: done

**Pull request merged**